dimod/constrained.py

```diff
@@ -1959,6 +1959,35 @@ def var_encoder(v):
             return f'{vartype_name[self.vartype(v)]}({v!r})'
 
         sio = StringIO()
+
+        def render_limited_number(iterable, render_element):
```

Comment:
@arcondello What do you think of this function?

I'm not 100% happy with it.  I feel there is a better way, some `itertools` or other simpler solution.  But such hasn't occurred to me yet.

Reply:
Nothing jumps out at me either